### PR TITLE
Adds Support for Yarn Install If Supported

### PR DESCRIPTION
**Pull request by benmarten**
checks if yarn is available and runs yarn install on setup. falls back to npm if yarn is not
installed.

- [x] I have read the [Contributing Documents](https://github.com/DaftMonk/generator-angular-fullstack/blob/master/contributing.md)
- [x] My commit(s) follow the [AngularJS commit message guidelines](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/)
- [x] The generator's tests pass (`generator-angular-fullstack$ npm test`)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -148,7 +148,9 @@ gulp.task('installFixtures', function() {
     }, 1 * 1000);
     shell.cd('test/fixtures');
 
-    execAsync('npm install --quiet', {cwd: '../fixtures'}).then(() => {
+    execAsync('type yarn &> /dev/null | yarn install || npm install --quiet', {
```
Comment (Awk34) [inline gulpfile.js]: we should be checking if the user is on Windows, and using the proper Windows commands if so

Comment (benmarten) [inline gulpfile.js]: I'm not on windows and don't have access to it atm.
I found this: https://superuser.com/questions/175466/determine-if-command-is-recognized-in-a-batch-file
Hence we could use: `WHERE yarn >nul 2>nul; IF %ERRORLEVEL% EQU 0 yarn install ELSE npm install --quiet;`
Can you or sb. confirm that this works please?

Comment (Awk34) [inline gulpfile.js]: This worked for me:

```js
yarn --version >nul 2>&1 && ( yarn install ) || ( npm install --quiet )
```

We should be able to use [process.platform](https://nodejs.org/api/process.html#process_process_platform) to figure out what the user is on.

Comment (benmarten) [inline gulpfile.js]: Updated. Tested regular install on macOS.
Tried running:  `gulp installFixtures` but that doesn't work.

Can you test on Windows too please?

**Comment by Awk34** [comment]
The Windows check fails, since `indexOf('win')` on `'win32'` is `0`. Just check `process.platform === 'win32'` instead

**Comment by benmarten** [comment]
Gotta, I thought there might be win64, changed it to win32 now though ;)

**Comment by Awk34** [comment]
Yeah, that would make sense, but the [process docs](https://nodejs.org/dist/latest-v7.x/docs/api/process.html) only has `win32`. There's a different `arch` property for 32 vs 64 bit.

Anyhow, your code will still not work on Windows, since `process.platform.indexOf('win32')` on Windows will result in `0`, which is falsey and will not enter the `if` block.

You can either use `process.platform.indexOf('win32') >= 0`, or `process.platform === 'win32'`

**Comment by benmarten** [comment]
updated.

**Pull request merged**